server/s3_bucket_metadata_v1.cc

```diff
@@ -482,12 +469,17 @@ void S3BucketMetadataV1::save_replica() {
 void S3BucketMetadataV1::save_bucket_info_failed() {
   s3_log(S3_LOG_INFO, request_id, "Entering\n");
   s3_log(S3_LOG_ERROR, request_id, "Saving of Bucket metadata failed\n");
-  if (motr_kv_writer->get_state() == S3MotrKVSWriterOpState::failed_to_launch) {
-    state = S3BucketMetadataState::failed_to_launch;
+  auto kv_state = motr_kv_writer->get_state();
+  if (!should_cleanup_global_idx) {
+    if (kv_state == S3MotrKVSWriterOpState::failed_to_launch) {
+      state = S3BucketMetadataState::failed_to_launch;
+    } else {
+      state = S3BucketMetadataState::failed;
+    }
+    this->handler_on_failed();
   } else {
-    state = S3BucketMetadataState::failed;
```

Comment:
will this state value will be used at other places

Reply:
yes, this is an operation status